src/partition/sph-partition-matches.cc

```diff
@@ -160,54 +217,124 @@ Worker::Worker(ConfigStore const& config)
+            std::pair<int32_t, int32_t> chunkSubChunk1;
+            bool null1 = _editor.isNull(_id1Field);
+            if (!null1) {
+                id1 = _editor.get(_id1Field, true);
+                try {
+                    chunkSubChunk1 = _objectIndex1->read(id1);
+                } catch (std::out_of_range const&) {
+                    std::string const msg = "No entry for identifier " + _id1FieldName + "=" + id1 +
+                                            " was found in the index map.";
+                    if (_abortOnMissingId1) throw std::runtime_error(msg);
+                    std::cerr << msg << std::endl;
+                    null1 = true;
+                }
+            }
+            std::string id2;
+            std::pair<int32_t, int32_t> chunkSubChunk2;
+            bool null2 = _editor.isNull(_id2Field);
             if (!null2) {
-                // Both positions are valid.
-                if (angSep(cartesian(sc1), cartesian(sc2)) * DEG_PER_RAD >
-                    _chunker.getOverlap() - EPSILON_DEG) {
-                    throw std::runtime_error(
-                            "Partitioning positions in match "
-                            "record are separated by more "
-                            "than the overlap radius.");
+                id2 = _editor.get(_id2Field, true);
+                try {
+                    chunkSubChunk2 = _objectIndex2->read(id2);
+                } catch (std::out_of_range const&) {
+                    std::string const msg = "No entry for identifier " + _id2FieldName + "=" + id2 +
+                                            " was found in the index map.";
+                    if (_abortOnMissingId2) throw std::runtime_error(msg);
+                    std::cerr << msg << std::endl;
+                    null2 = true;
                 }
-                if (loc1.chunkId == loc2.chunkId) {
-                    // Both positions are in the same chunk.
-                    _editor.set(_flagsField, '3');
-                    silo.add(loc1, _editor);
-                    continue;
+            }
+            if (null1 && null2) {
+                throw std::runtime_error(
+                        "Values of both identifiers in a match record are set to NULLs, or "
+                        "no entries for both identifiers were found in the index maps.");
+            }
+            if (!null1) {
+                int32_t const chunkId1 = chunkSubChunk1.first;
+                int32_t const subChunkId1 = chunkSubChunk1.second;
+                _editor.set(_chunkIdField, chunkId1);
+                _editor.set(_subChunkIdField, subChunkId1);
+                if (!null2) {
+                    // Both positions are valid.
+                    int32_t const chunkId2 = chunkSubChunk2.first;
+                    if (chunkId1 == chunkId2) {
+                        // Both positions are in the same chunk.
+                        _editor.set(_flagsField, '3');
```

Comment:
Yeah, I had a look for that the other day, because the meaning/use of the flag column was unfamiliar to me.  I didn't see anywhere where it was actually used/referenced beyond the partitioner.  I think it is likely dead wood.